ChainRouter.php

```diff
@@ -291,8 +291,19 @@ private function rebuildRequest($uri)
                 $server['HTTP_HOST'] .= ':'.$this->context->getHttpPort();
             }
         }
+        // use the array field to have both domain and port.
+        if (isset($server['HTTP_HOST'])) {
+            $pathinfo = $server['HTTP_HOST'].$pathinfo;
+        } else {
+            $pathinfo = 'localhost'.$pathinfo;
```

Comment:
do we want to do this? and if we do, should we do the same for the case without RequestContext being set at all?